src/components/KMultiselect/KMultiselect.vue

```diff
@@ -856,11 +907,23 @@ watch(filteredItems, () => {
 // watch for programmatic changes to model
 watch(value, (newVal, oldVal) => {
   if (JSON.stringify(newVal) !== JSON.stringify(oldVal)) {
-    const items = unfilteredItems.value.filter((item: MultiselectItem) => newVal.includes(item.value))
-    if (items.length) {
-      handleMultipleItemsSelect(items)
-    } else if (!newVal.length) {
+    if (!newVal.length) {
       clearSelection()
+      return
+    }
+
+    const previouslySelectedItems = new Set<string>(oldVal)
+    const currentlySelectedItems = new Set<string>(newVal)
+
+    const selectedAndPresentItems = unfilteredItems.value.filter((item: MultiselectItem) => currentlySelectedItems.has(item.value))
+    const deselectedItems = selectedItems.value.filter((item: MultiselectItem) => !currentlySelectedItems.has(item.value) && previouslySelectedItems.has(item.value))
+
+    if (deselectedItems.length) {
+      handleMultipleItemDeselect(deselectedItems)
```

Comment:
```suggestion
      handleMultipleItemsDeselect(deselectedItems)
```

see the comment above